### PR TITLE
Switch oxide to binary

### PR DESCRIPTION
**Pull request by Eeems**
Toltec builds are crashing on device.

### CODE REVIEW

**Comment by Eeems** [comment]
@matteodelabre FYI I had to switch to a binary release because the build output of the toolchain is crashing.

**Comment by matteodelabre** [comment]
I think this indicates a problem either in the toolchain or in oxide, which I’d rather fix instead of working around.

**Comment by matteodelabre** [comment]
When compiling oxide in debug mode, I get no crash, which makes it harder to see what’s going on. The crash I get on release mode (“terminate called without an active exception”) might indicate that you’re destructing a thread without joining it first.

**Comment by matteodelabre** [comment]
I’m able to reproduce the crash with debug symbols when compiling in release mode with forced debug symbols. Seems that the crash happens in the constructor of AppsAPI. But it could come from another thread, and I’m not able to switch threads with gdb.

**Review by raisjn — APPROVED** [review]
> I think this indicates a problem either in the toolchain or in oxide, which I’d rather fix instead of working around.

agree, but there are people who have broken builds, so accepting for now while debugging happens

**Comment by Eeems** [comment]
> I’m able to reproduce the crash with debug symbols when compiling in release mode with forced debug symbols. Seems that the crash happens in the constructor of AppsAPI. But it could come from another thread, and I’m not able to switch threads with gdb.

Could you open an issue on oxide about this? I'm surprised that this doesn't happen with to rM toolchain.

File: package/oxide/package [hunk 1]
```diff
@@ -2,24 +2,20 @@
 pkgname=oxide
 pkgdesc="Launcher application"
 url=https://github.com/Eeems/oxide
-pkgver=2.0~beta~1
+pkgver=2.0~beta~2
```
Comment (matteodelabre) [inline package/oxide/package:5]: The `pkgrel` part of the package version should be separated from the rest using a dash, not a tilde, i.e. `2.0~beta-2` instead of `2.0~beta~2`. Otherwise, if a version is released that drops the `~beta` suffix, we get `2.0~2`, which is actually sorted lower than `2.0~beta~2`. You can use `opkg compare-versions` to test if a version is sorted lower or higher than another.

File: package/oxide/package [hunk 2]
```diff
@@ -2,24 +2,20 @@
...
-    make release
-}
+source=(https://github.com/Eeems/oxide/releases/download/v2.0-beta/oxide.zip)
+sha256sums=(d3bac44b96c78726fcbfbd8f7a23d3bd4e4ae3f1f794714190100ebe602a8433)
 
 package() {
-    cp -ar "$srcdir"/release/* "$pkgdir"
+    unzip "$srcdir"/oxide.zip -d "$pkgdir"
+
+    tree "$pkgdir"
```
Comment (matteodelabre) [inline package/oxide/package:18]: Why is this necessary? `tree` is already called [in the packaging script](https://github.com/toltec-dev/toltec/blob/249e558582c88ca66194d8acf53738500c2266c6/scripts/package-build#L146).

**Review by matteodelabre — COMMENTED** [review]
Some little things I didn’t catch before the PR was merged.

**Comment by matteodelabre** [comment]
> > I’m able to reproduce the crash with debug symbols when compiling in release mode with forced debug symbols. Seems that the crash happens in the constructor of AppsAPI. But it could come from another thread, and I’m not able to switch threads with gdb.
> 
> Could you open an issue on oxide about this? I'm surprised that this doesn't happen with to rM toolchain.

I’m surprised too. The gcc version in the reMarkable toolchain is 7.3.0, and the gcc in the toltec images is 8.3.0, which might explain the difference. I’ll open an issue.

File: package/oxide/package (hunk 2 above)
Comment (Eeems) [reply package/oxide/package:18]: If you look at the history you'll see why it happened. The code that handles extracting the zip automatically does some funky things about assuming that you don't care about the top level directory in the zip file.

Comment (matteodelabre) [reply package/oxide/package:18]: I see. I tried to maintain consistency with Arch’s makepkg while implementing it, so the behavior should be roughly the same. Do you have any suggestion on how to improve the extracting part of the building script?

Comment (Eeems) [reply package/oxide/package:18]: I'd have to look at what you are doing, but at the very least we should document the behaviour.
Ideally we should only extract assuming the zip contains a single directory if that's actually the case.
